### PR TITLE
Bug 2080687: request prometheus service account token

### PR DESCRIPTION
**Pull request by ibihim**
## What

Make a call to token request API to get a token.

## Why

Don't rely on auto-generated secrets anymore:

- https://bugzilla.redhat.com/show_bug.cgi?id=2080687
- https://github.com/kubernetes/kubernetes/pull/108309/files

Signed-off-by: Krzysztof Ostrowski <kostrows@redhat.com>

### CODE REVIEW

File: test/extended/router/metrics.go [hunk 1]
```diff
@@ -83,10 +84,21 @@ var _ = g.Describe("[sig-network][Feature:Router]", func() {
...
 
-		// Extract a bearer token from Prometheus authorized to access
-		// the router metrics URL.
-		bearerToken, err = findMetricsBearerToken(oc)
+		token, err := oc.AdminKubeClient().
+			CoreV1().
+			ServiceAccounts("openshift-ingress").
+			CreateToken(
+				context.TODO(),
+				"router-metrics-sa",
```
Comment (s-urbaniak) [inline test/extended/router/metrics.go:92]: this should be `router`

File: test/extended/router/metrics.go [hunk 2]
```diff
@@ -83,10 +84,21 @@ var _ = g.Describe("[sig-network][Feature:Router]", func() {
 		o.Expect(err).NotTo(o.HaveOccurred())
 		username, password = string(statsSecret.Data["statsUsername"]), string(statsSecret.Data["statsPassword"])
 
-		// Extract a bearer token from Prometheus authorized to access
-		// the router metrics URL.
-		bearerToken, err = findMetricsBearerToken(oc)
+		token, err := oc.AdminKubeClient().
+			CoreV1().
+			ServiceAccounts("openshift-ingress").
```
Comment (s-urbaniak) [inline test/extended/router/metrics.go:89]: this should be `openshift-monitoring`

File: test/extended/router/metrics.go (hunk 1 above)
Comment (s-urbaniak) [inline test/extended/router/metrics.go:92]: this should be `prometheus-k8s`

**Comment by s-urbaniak** [comment]
/retitle Bug: 2080687: request prometheus service account token

**Review by soltysh — APPROVED** [review]
/lgtm
/approve

File: test/extended/router/metrics.go [hunk 3]
```diff
@@ -83,10 +84,21 @@ var _ = g.Describe("[sig-network][Feature:Router]", func() {
...
-		bearerToken, err = findMetricsBearerToken(oc)
+		token, err := oc.AdminKubeClient().
+			CoreV1().
+			ServiceAccounts("openshift-monitoring").
+			CreateToken(
+				context.TODO(),
+				"prometheus-k8s",
+				&authenticationv1.TokenRequest{
+					Spec: authenticationv1.TokenRequestSpec{
+						Audiences: []string{"https://kubernetes.default.svc"},
```
Comment (alvaroaleman) [inline test/extended/router/metrics.go:95]: @ibihim @tkashem @s-urbaniak @soltysh changes like this one break on all clusters that do not have `kubernetes.default.svc` for the KAS SA account issuer configured, eg when using AWS STS. 

I am fixing this particlar instance in https://github.com/openshift/origin/pull/27120 but I would greatly appreciate it if you are mindful of this when reviewing future PRs.

**Pull request merged**